src/printer.js

```diff
@@ -1442,8 +1442,11 @@ function genericPrintNoParens(path, options, print) {
       ]);
     case "DeclareInterface":
     case "InterfaceDeclaration": {
-      const parent = path.getParentNode(1);
-      if (parent && parent.type === "DeclareModule") {
+      const nodeSource = options.originalText.slice(0, util.locStart(n));
+      if (
+        n.type === "DeclareInterface" ||
+          (options.parser === "flow" && nodeSource.match(/declare\s*$/))
```

Comment:
No, declare is the only thing that we have issues with. It there are more in the future we may want to generalize but that's unlikely.